File: templates/python/control_mpc_stage_step.py
"""control_mpc_stage_step: MPC 控制进阶阶段阶梯曲线（step 模式，合成数据）."""
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..', '_utils', 'python'))
from generated_patterns import make_template_figure


def make_figure(title='advanced MPC control: stage step curve'):
    return make_template_figure('step_curve', seed=1617, title=title, domain='advanced MPC control', topic='stage step curve')


if __name__ == '__main__':
    fig = make_figure()
    fig.savefig(__file__.replace('.py', '.png'), dpi=150)

File: templates/matlab/control_mpc_stage_step.m
function fig = control_mpc_stage_step()
    addpath(fullfile(fileparts(mfilename('fullpath')),'..','..','_utils','matlab'));
    fig = generated_template_figure('step_curve', 1617, 'advanced MPC control: stage step curve', 'advanced MPC control', 'stage step curve');
end
